lib/puppet/provider/user/directoryservice.rb

```diff
@@ -562,6 +576,17 @@ def get_shadow_hash_data(users_plist)
     end
   end
 
+  # This method will check if authentication_authority key of a user's plist
+  # contains SALTED_SHA512_PBKDF2
+  def has_sha512_pbkdf2_authentication_authority_set?(users_plist)
+    if (authority = users_plist['authentication_authority'])
```

Comment:
Puppet code is not a big fan of assignment inside conditions, recently there was a refactoring that removed this kind of code. Would be better to be consistent with that style: assign -> check -> use